libs/wingc/src/type_check.rs

```diff
@@ -2577,26 +2591,44 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
-	/// Validate if the fields of a class are initialized in the constructor (init).
+	/// Validate if the fields of a class are initialized in the preflight constructor (init).
 	///
 	/// # Arguments
 	///
-	/// * `statements` - The constructor scope (init)
+	/// * `scope` - The constructor scope (init)
 	/// * `fields` - All fields of a class
+	/// * `phase` - initializer phase
 	///
-	fn check_class_field_initialization(&mut self, scope: &Scope, fields: &[ClassField]) {
+	fn check_class_field_initialization(&mut self, scope: &Scope, fields: &[ClassField], phase: Phase) {
 		let mut visit_init = VisitClassInit::default();
 		visit_init.analyze_statements(&scope.statements);
 		let initialized_fields = visit_init.fields;
 
+		let (current_phase, forbidden_phase) = if phase == Phase::Inflight {
+			("Inflight", Phase::Preflight)
+		} else {
+			("Preflight", Phase::Inflight)
+		};
+
 		for field in fields.iter() {
+			let result = initialized_fields.iter().find(|&s| &s.name == &field.name.name);
 			// inflight or static fields cannot be initialized in the initializer
-			if field.phase == Phase::Inflight || field.is_static {
+			if field.phase == forbidden_phase || field.is_static {
+				if let Some(result) = result {
```

Comment:
I would shuffle this around a little bit to avoid some additional nesting:

```rust
let matching_field = initialized_fields.iter().find(|&s| &s.name == &field.name.name);
if field.phase != forbidden_phase && !field.is_static {
	continue;
}

if let Some(matching_field) = matching_field {
  self.type_error(...);
} else {
  self.type_error(...);
}
```

Reply:
@Chriscbr the `{} cannot be initialized in the {} initializer` needs before the `continue`

What I can do is remove the `if let Some(matching_field) = matching_field { ... }` and create a function, then I can call when is needed

Reply:
Ah gotcha. I'm okay with leaving this as is just as well 👍

Reply:
@Chriscbr to create the `{} cannot be initialized in the {} initializer` I need to create the validation if matching_field exist before the `continue`

This change breaks the validation